src/magic.cpp

```diff
@@ -31,7 +31,9 @@ const std::map<std::string, valid_target> target_map = {
     { "hostile", valid_target::target_hostile },
     { "self", valid_target::target_self },
     { "ground", valid_target::target_ground },
-    { "none", valid_target::target_none }
+    { "none", valid_target::target_none },
+    { "item", valid_target::target_item },
+    { "creature", valid_target::target_creature }
```

Comment:
what's the point of this? you can just add "hostile" "ally" and "self" and it'll cover this.

Reply:
It supposed to be "creature" = "ally" + "hostile" + "self". From my warcraft3 modding experience, overlapping flags were very convenient for modding. Though they are harder to code. I feel like there will be much more flags soon, and it can be troublesome to specify every particular one in a json.

Reply:
I do like the idea of moving objects, especially things like equipment and vehicles.

Reply:
i have no problems with the item flag. i just think the overlapping flag is redundant, and can become confusing.

Reply:
Removed 'creature' flag for now